### DISCUSSION TITLE
v4.0.40 pre-release

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#100] [General]
An enhancement, a fix, and a tweak.

Fix: @Unlik hit a snag with his HA deployment type.
Enhancement: More half hour interpolation goodness for some sensors.
Tweak: @gcoan enhanced his sample chart

As always, both positive and negative feedback is most welcome, and this will contribute to getting the release ultimately published as latest. Report back, and don't be shy!

Feedback in the past has been gratefully received from @dtswk, @EddieSpaghettie, @ffm777, @gcoan, @hwb45, @iainfogg, @isorin, @jeffb1974, @kernow, @ladaan, @Markyt77, @nilogax, @Noodleyman, @ProphetOfDoom, @rholligan, @Unlik and @wimdebruyn, and continued thanks to them for their interest in beta releases (and alphas!) and for their contributions.

> [!NOTE]
> The BJReplay fork is **_still_** not published for searching in HACS, so continued use of a manual install in HACS is the go. As explained earlier we're going to keep any releases as pre-release on the fork until that happens, as any 'latest' version change would require modification of that HACS pull request.

## What's Changed
* Interpolated forecast 0/30/60 power and energy X hours by @autoSteve
* Ensure config directory is always relative to install location by @autoSteve
* Sample PV chart enhancements by @gcoan

**Full Changelog**: https://github.com/BJReplay/ha-solcast-solar/compare/v4.0.39...v4.0.40

**Comment by iainfogg** [comment]
I've installed, I'll shout if there are any issues

**Comment by wimdebruyn** [comment]
Looking good so far

**Comment by autoSteve** [comment]
Well, this is a wild release, clearly! Closing and moving on to v4.0.41...
